### PR TITLE
feat(ColorPicket): create color-picker component for new Theme page

### PR DESCRIPTION
**Pull request by bityutskiyAO**

### CODE REVIEW

File: src/blocks/Examples/components/Showcase/Showcase.tsx [hunk 1]
```diff
@@ -105,6 +106,11 @@ export const Showcase: React.FC<ShowcaseProps> = ({color, theme}) => {
                                     <Switch />
                                 </div>
                             </div>
+                            <div className={b('col-item')}>
+                                <div className={b('col-item-elem')}>
+                                    <ColorPickerInput defaultValue="#000000" />
+                                </div>
+                            </div>
```
Comment (bityutskiyAO) [inline src/blocks/Examples/components/Showcase/Showcase.tsx:113]: i will remove it after getting approve from design review

File: src/components/ColorPickerInput/ColorPickerInput.tsx [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+                errorMessage={errorMessage}
+                validationState={validationError}
+                view="normal"
+                size="l"
+                onChange={onChange}
+                startContent={<ColorPreview color={color} />}
+                endContent={
+                    <Button
+                        view="flat-action"
+                        className={b('select-wrapper')}
```
Comment (imsitnikov) [inline src/components/ColorPickerInput/ColorPickerInput.tsx:88]: Is it used?

**Review by imsitnikov — APPROVED** [review]

File: src/components/ColorPickerInput/ColorPickerInput.tsx (hunk 2 above)
Comment (bityutskiyAO) [reply src/components/ColorPickerInput/ColorPickerInput.tsx]: removed

File: src/components/ColorPickerInput/utils.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+export const hexRegexp = /^#[a-fA-F0-9]{6}$/;
```
Comment (dgaponov) [inline src/components/ColorPickerInput/utils.ts:1]: All regex's in your code are used with global flag. Let's add it to all the regulars and use it like this:

```
export const hexRegexp = /^#[a-fA-F0-9]{6}$/g;

new RegExp(hexRegexp, 'g').test(value) -> hexRegexp.test(value)
```

File: src/components/ColorPickerInput/ColorPickerInput.tsx [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+    const [validationError, setValidationError] = useState<TextInputProps['validationState']>();
+
+    const colorInputRef = useRef<HTMLInputElement>(null);
+
+    const managedValue = value || inputValue;
+
+    const onChange: ChangeEventHandler<HTMLInputElement> = useCallback(
+        (event) => {
+            const newValue = event.target.value.replaceAll(' ', '');
+            onChangeExternal?.(newValue);
```
Comment (dgaponov) [inline src/components/ColorPickerInput/ColorPickerInput.tsx:43]: Can we trigger `onChangeExternal` only for valid value?

File: src/components/ColorPickerInput/ColorPickerInput.tsx [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+            }
+        },
+        [onChangeExternal],
+    );
+
+    const onNativeInputChange: ChangeEventHandler<HTMLInputElement> = useCallback((e) => {
+        const newValue = e.target.value.toUpperCase();
+
+        setColor(newValue);
+        setInputValue(newValue);
```
Comment (dgaponov) [inline src/components/ColorPickerInput/ColorPickerInput.tsx:71]: It seems that the validation status needs to be reset here.

File: src/components/ColorPickerInput/utils.ts (hunk 3 above)
Comment (bityutskiyAO) [reply src/components/ColorPickerInput/utils.ts:1]: it not working properly, check  https://learn.javascript.ru/regexp-methods

P.S. If i use  `regexp.test` more than one it's get wrong result

File: src/components/ColorPickerInput/ColorPickerInput.tsx (hunk 4 above)
Comment (bityutskiyAO) [reply src/components/ColorPickerInput/ColorPickerInput.tsx:43]: In my head `onChangeExternal`  is used for controlling input not by component, but for parent who use it. It should update value on every changing

File: src/components/ColorPickerInput/ColorPickerInput.tsx (hunk 5 above)
Comment (bityutskiyAO) [reply src/components/ColorPickerInput/ColorPickerInput.tsx:71]: I will add it in next task (or u can add it by yourself)

**Pull request merged**